server/rest/src/main/java/org/infinispan/rest/NettyRestServer.java

```diff
@@ -0,0 +1,98 @@
+      return createServer(configuration, new DefaultCacheManager(), EmbeddedCacheManager::stop);
+   }
+
+   public static NettyRestServer createServer(RestServerConfiguration configuration, String configurationFile) {
+      return createServer(configuration, createCacheManager(configurationFile), EmbeddedCacheManager::stop);
+   }
+
+   public static NettyRestServer createServer(RestServerConfiguration configuration, EmbeddedCacheManager manager) {
+      return createServer(configuration, manager, cm -> { });
+   }
+
+   public static NettyRestServer createServer(RestServerConfiguration configuration, EmbeddedCacheManager manager,
+                                              Consumer<? super EmbeddedCacheManager> consumer) {
+      // Start caches first, if not started
+      startCaches(manager);
+
+      NettyJaxrsServer netty = new NettyJaxrsServer();
+      ResteasyDeployment deployment = new ResteasyDeployment();
+      netty.setDeployment(deployment);
+      netty.setHostname(configuration.host());
+      netty.setPort(configuration.port());
+      netty.setRootResourcePath("");
+      netty.setSecurityDomain(null);
+      return new NettyRestServer(manager, configuration, netty, consumer);
+   }
+
+   private static void startCaches(EmbeddedCacheManager cm) {
+      // Start defined caches to avoid issues with lazily started caches
+      cm.getCacheNames().forEach(name -> SecurityActions.getCache(cm, name));
+
+      // Finally, start default cache as well
+      cm.getCache();
+   }
+
+   private static EmbeddedCacheManager createCacheManager(String cfgFile) {
+      try {
+         return new DefaultCacheManager(cfgFile);
+      } catch (IOException e) {
+         log.errorReadingConfigurationFile(e, cfgFile);
+         return new DefaultCacheManager();
+      }
+   }
+
+   private NettyRestServer(EmbeddedCacheManager cacheManager, RestServerConfiguration configuration, NettyJaxrsServer netty,
+                   Consumer<? super EmbeddedCacheManager> onStop) {
+      this.cacheManager = cacheManager;
+      this.configuration = configuration;
+      this.netty = netty;
+      this.onStop = onStop;
+   }
+
+   @Override
+   public void start() {
+      netty.start();
+      ResteasyDeployment deployment = netty.getDeployment();
+      configuration.getIgnoredCaches().forEach(this::ignoreCache);
+      RestCacheManager restCacheManager = new RestCacheManager(cacheManager, this::isCacheIgnored);
+      Server server = new Server(configuration, restCacheManager);
+      deployment.getRegistry().addSingletonResource(server);
+      deployment.getProviderFactory().register(new RestAccessLoggingHandler(), ContainerRequestFilter.class,
+         ContainerResponseFilter.class);
+      log.startRestServer(configuration.host(), configuration.port());
+   }
+
+   @Override
+   public void stop() {
+      netty.stop();
+      onStop.accept(cacheManager);
+   }
+}
```

Comment:
That rule is not yet enabled on master, but it will soon be (https://github.com/infinispan/infinispan/pull/4537).